primitives/payment-handler/src/tests.rs

```diff
@@ -179,17 +181,58 @@ mod asset_fee_handler {
 			});
 		}
 
+		#[test]
+		fn withdraw_and_pay_fee_deposits_to_tournament_affiliates_and_treasury_works_with_vouchers()
+		{
+			ExtBuilder::default().vouchers(&[(ALICE, 5)]).build().execute_with(|| {
+				let fee = 2;
+				let alice_balance_before = Assets::balance(WHITELISTED_ASSET_ID, ALICE);
+				let fee_beneficiary = FERDIE;
+
+				let alice_initial_vouchers = VOUCHERS
+					.with_borrow(|voucher_store| voucher_store.get(&ALICE).copied())
+					.expect("Should contain remaining vouchers");
+
+				TestAssetFeeHandler::withdraw_and_pay_fees(
+					&ALICE,
+					VOUCHER_ASSET_PAYMENT,
+					fee,
+					&TournamentFeeId::Paying,
+					&AffiliateFeeId::Paying,
+					&fee_beneficiary,
+				)
+				.unwrap();
+
+				// check that the fee has not been deducted from the payer
+				assert_eq!(Assets::balance(WHITELISTED_ASSET_ID, ALICE), alice_balance_before);
+
+				// check that no tournament allocation has occurred
+				assert_eq!(Assets::balance(WHITELISTED_ASSET_ID, TOURNAMENT_TREASURY), 0);
+
+				// check that no fee has also been allocated to the affiliates
+				assert_eq!(Assets::balance(WHITELISTED_ASSET_ID, BOB), 0);
+				assert_eq!(Assets::balance(WHITELISTED_ASSET_ID, CHARLIE), 0);
+				assert_eq!(Assets::balance(WHITELISTED_ASSET_ID, DAVE), 0);
+
+				// check that the requested vouchers have been deducted from storage
+				let alice_current_vouchers = VOUCHERS
+					.with_borrow(|voucher_store| voucher_store.get(&ALICE).copied())
+					.expect("Should contain remaining vouchers");
+				assert_eq!(alice_current_vouchers, alice_initial_vouchers - fee);
```

Comment:
I think that the implementation should be: 1 Voucher == 1 Execution. The production implementation has no conversion from the fee to the vouchers, as there exists no market. So my proposition is the only way it makes sense to me.